src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SSRP.cs

```diff
@@ -372,5 +396,40 @@ internal static string SendBroadcastUDPRequest()
             }
             return response.ToString();
         }
+
+        private static void SplitIPv4AndIPv6(IPAddress[] input, out IPAddress[] ipv4Addresses, out IPAddress[] ipv6Addresses)
+        {
+            ipv4Addresses = Array.Empty<IPAddress>();
+            ipv6Addresses = Array.Empty<IPAddress>();
+
+            if (input != null && input.Length > 0)
```

Comment:
Maybe revert this condition and have fast exit? Code above looks a bit unnatural with this condition
```
ipv4Addresses = Array.Empty<IPAddress>();
ipv6Addresses = Array.Empty<IPAddress>();
```

Reply:
It's like this because they're out parameters. I suppose it's a personal style choice but I always assign the default values for out parameters at the start of the method so that anywhere inside the body I can easily return without worrying about whether they're assigned or not.